### ISSUE TITLE
Types not working in V2.5.0

### ISSUE THREAD

**Issue by hari9973** [fluent/fluent-plugin-grok-parser#66]
In previous version(ie 2.4.0) i used the below config:-

```
<parse>
   @type grok
   types log_timestamp:time:%d/%b/%Y:%H:%M:%S
   grok_pattern %{IPORHOST:remote_addr} - %{HTTPDUSER:remote_user} \[%{HTTPDATE:log_timestamp}\] "(?:%{WORD:http_verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:http_version})?|%{DATA:rawrequest})" %{NUMBER:http_response_code} %{NUMBER:bytes:integer}( %{NUMBER:reqtime}| -)?( %{NUMBER:upstime}| -)?( %{HOSTPORT:upstream_addr} | - )?(?<pipe>[pP]|.)? %{QS:http_referer} %{QS:http_user_agent}
 </parse>
```

And got the output as below:-

```
{"remote_addr":"103.21.16.20","remote_user":"-","log_timestamp":1549156652000,"http_verb":"POST","request":"/game","http_version":"1.1","http_response_code":"200","bytes":"101","http_referer":"\"-\"","http_user_agent":"\"Python-urllib/2.7\"","log_file_path":"/home/access.log"}
```
But this does not seem to be working in the current version.can you provide me a solution

Thanks

Note:- I tried the following in new version and i am getting nil
` \[%{HTTPDATE:log_timestamp:time:%d/%b/%Y:%H:%M:%S %z}\]`

**Comment by okkez** [comment]
Do you use `in_tail` as an input plugin?
Could you show me your raw logs?

**Comment by hari9973** [comment]
This is my full source config:-
```
<source>
 @type tail
 tag nginx
 path /var/log/nginx/access.log
 pos_file /var/lib/td-agent/nginx.pos
 path_key log_file_path
 <parse>
   @type grok
   types log_timestamp:time:%d/%b/%Y:%H:%M:%S %z
   grok_pattern %{IPORHOST:remote_addr} - %{HTTPDUSER:remote_user} \[%{HTTPDATE:log_timestamp}\] "(?:%{WORD:http_verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:http_version})?|%{DATA:rawrequest})" %{NUMBER:http_response_code} %{NUMBER:bytes:integer}( %{NUMBER:reqtime}| -)?( %{NUMBER:upstime}| -)?( %{HOSTPORT:upstream_addr} | - )?(?<pipe>[pP]|.)? %{QS:http_referer} %{QS:http_user_agent}
 </parse>
</source>
```
Here are some raw logs:-
```
127.0.0.1 - - [03/Feb/2019:06:47:21 +0530] "POST /game HTTP/1.1" 200 101 "-" "Python-urllib/2.7"
127.0.0.1 - - [03/Feb/2019:06:47:22 +0530] "POST /game HTTP/1.1" 200 101 "-" "Python-urllib/2.7"
127.0.0.1 - - [03/Feb/2019:06:49:02 +0530] "POST /authentication HTTP/1.1" 200 145 "-" "Python-urllib/2.7"
```

**Comment by okkez** [comment]
Please trye #67 and following parse section.

```
  <parse>
    @type grok
    grok_pattern %{IPORHOST:remote_addr} - %{HTTPDUSER:remote_user} \[%{HTTPDATE:log_timestamp:time:%d/%b/%Y:%H:%M:%S %z}\] "(?:%{WORD:http_verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:http_version})?|%{DATA:rawrequest})" %{NUMBER:http_response_code} %{NUMBER:bytes:integer}( %{NUMBER:reqtime}| -)?( %{NUMBER:upstime}| -)?( %{HOSTPORT:upstream_addr} | - )?(?<pipe>[pP]|.)? %{QS:http_referer} %{QS:http_user_agent}
  </parse>
```

**Comment by hari9973** [comment]
Thank you for your help i made the changes to my local gem as in [#67](https://github.com/fluent/fluent-plugin-grok-parser/pull/67) and it is working as expected

**Comment by okkez** [comment]
Thank you for confirming.
Merged #67 and released v2.5.1.
